src/plugins/crypto/gpg.c

```diff
@@ -166,16 +197,39 @@ static int getGpgBinary (char ** gpgBin, KeySet * conf, Key * errorKey)
 	}
 
 
-	// last resort
-	const size_t defaultLen = strlen (ELEKTRA_CRYPTO_DEFAULT_GPG_BIN) + 1;
-	*gpgBin = elektraMalloc (defaultLen);
+	// last resort number one - check for gpg2 at /usr/bin/gpg2
+	// NOTE this might happen if the PATH variable is empty
+	*gpgBin = strdup (ELEKTRA_CRYPTO_DEFAULT_GPG2_BIN);
+	if (!(*gpgBin))
+	{
+		ELEKTRA_SET_ERROR (87, errorKey, "Memory allocation failed");
+		return -1;
+	}
+
+	if (isExecutable (*gpgBin, NULL) == 0)
+	{
+		return 1;
```

Comment:
No, `gpgBin` has to be freed by the caller. It gets deallocated only if an error occurs. Please see the doc of the method:

	/**
	 * @brief lookup the path to the gpg binary in conf.
	 * @param gpgBin holds allocated path to the gpg binary to be used or NULL in case of an error. Must bee freed by the caller.
	 * @param conf KeySet holding the plugin configuration.
	 * @param errorKey holds an error description if something goes wrong.
	 * @retval 1 on success.
	 * @retval -1 on error. In this case errorkey holds an error description.
	 */